modules/test/playwright/pages/calendar-web/CalendarWidgetPage.ts

```diff
@@ -0,0 +1,90 @@
+/** allDayCheckbox
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+export class CalendarWidgetPage {
+	readonly addEventButton: Locator;
+	readonly allDayCheckbox: Locator;
+	readonly calendarWidget: Locator;
+	readonly closeConfigurationButton: Locator;
+	readonly configurationMenuItem: Locator;
+	readonly endTime: Locator;
+	readonly publishEventButton: Locator;
+	readonly saveConfigurationButton: Locator;
+	readonly startTime: Locator;
+	readonly timeZoneDropdown: Locator;
+	readonly useGlobalTimeZoneCheckBox: Locator;
+
+	constructor(page: Page) {
+		this.addEventButton = page.getByRole('button', {name: 'Add Event'});
+		this.allDayCheckbox = page
+			.frameLocator('iframe')
+			.getByRole('checkbox', {
+				exact: true,
+				name: 'All Day',
+			});
+		this.calendarWidget = page.locator(
+			'.lfr-layout-structure-item-com-liferay-calendar-web-portlet-calendarportlet'
+		);
```

Comment:
@cgmonte as discussed, I wasn't able to get rid of this locator

Reply:
That's OK! Everytime I review a Playwright PR, I ask if the author tried to replace locators with Locators. I just want people to try. Many times it is possible to find a working Locator in few minutes. If this is not the case, just go for the regular locator, which you did here. Thanks